sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import fs from "fs-extra";
+import nise from "nise";
+import queryString from "query-string";
+import {
+  isBrowser,
+  blobToString,
+  escapeRegExp,
+  env,
+  TestInfo,
+  isPlaybackMode,
+  isRecordMode
+} from "./utils";
+import { customConsoleLog } from "./customConsoleLog";
+
+let nock: any;
+
+let replaceableVariables: { [x: string]: string } = {};
+export function setReplaceableVariables(a: { [x: string]: string }): void {
+  replaceableVariables = a;
+  if (isPlaybackMode()) {
+    // Providing dummy values to avoid the error
+    Object.keys(a).map((k) => {
+      env[k] = a[k];
+    });
+  }
+}
+
+let replacements: any[] = [];
+export function setReplacements(maps: any): void {
+  replacements = maps;
+}
+
+export function setEnviromentOnLoad() {
+  if (!isBrowser() && (isRecordMode || isPlaybackMode)) {
+    nock = require("nock");
+  }
+
+  if (isBrowser() && isRecordMode()) {
+    customConsoleLog();
+  }
+
+  // TODO - The following will be moved to the storage sdks when @azure/test-utils-recorder is imported
+  if (isPlaybackMode()) {
+    // Providing dummy values to avoid the error [ENVs for storage packages]
+    env.ACCOUNT_NAME = "fakestorageaccount";
+    env.ACCOUNT_KEY = "aaaaa";
+    env.ACCOUNT_SAS = "aaaaa";
+    env.STORAGE_CONNECTION_STRING = `DefaultEndpointsProtocol=https;AccountName=${
+      env.ACCOUNT_NAME
```

Comment:
Storage-specific functionality should be outside of the recorder.  I see the TODO, but it'd be good to make sure we're landing something general first, and then we can land PRs for storage and kv to use it.

Reply:
Done